include/dynd/types/string_type.hpp

```diff
@@ -22,6 +22,8 @@ class DYND_API string : public bytes {
 
   string(const std::string &other) : string(other.data(), other.size()) {}
 
+  string(const bytes &other) : string(other.data(), other.size()) {}
+
   bool operator<(const string &rhs) const
```

Comment:
It's to support:

```
dynd::string a("first");
dynd::string b("second");
dynd::string c(a + b);
```

since `operator+` returns bytes.

There might be a better way, of course.